crates/sparrow-runtime/src/key_hash_inverse.rs

```diff
@@ -255,8 +296,18 @@ impl ThreadSafeKeyHashInverse {
     /// This method is thread safe. It acquires the read lock to check if
     /// any of the keys need to be added to the inverse map, and only acquires
     /// the write lock if needed.
-    pub async fn add(&self, keys: ArrayRef, key_hashes: &UInt64Array) -> anyhow::Result<()> {
-        anyhow::ensure!(keys.len() == key_hashes.len());
+    pub async fn add(
```

Comment:
Is this method still safe? Why did we require a `blocking_add`?

Reply:
This one was async and most of the python code doesn't call from an async executor.